control/test_stand/generate_test_sequence.py

```diff
@@ -0,0 +1,153 @@
+            servo_0=cmd[0],
+            servo_1=cmd[1],
+            servo_2=cmd[2],
+            servo_3=cmd[3]
+        )
+
+    return text
+
+
+def write_file(text, path):
+    with open(path, 'w') as f:
+        f.write(text)
+
+
+def set_zero():
+    cmds = np.array([
+        [0.0, 0.0, 0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0],
+    ])
+
+    return cmds
+
+
+def sine_wave_all(freq_hz):
+    milliseconds = np.arange(0.0, LENGTH_TICKS)
+
+    seconds = milliseconds * 1e-3
+
+    angular_freq_radps = freq_hz / (2.0 * np.pi)
+
+    return MAX_ANGLE_RAD * np.vstack([
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds)
+    ]).transpose()
+
+
+def symmetric(included, freq_hz=3.0):
+    milliseconds = np.arange(0.0, LENGTH_TICKS)
+
+    seconds = milliseconds * 1e-3
+
+    angular_freq_radps = freq_hz / (2.0 * np.pi)
+
+    result = MAX_ANGLE_RAD * np.vstack([
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        -np.sin(angular_freq_radps * seconds),
+        -np.sin(angular_freq_radps * seconds)
+    ]).transpose()
+
+    not_incls = ALL_INDICES - set(included)
+    for not_incl in not_incls:
+        result[:, not_incl] == 0.0
+
+    return result
+
+
+def antisymmetric(included, freq_hz=3.0):
+    milliseconds = np.arange(0.0, LENGTH_TICKS)
+
+    seconds = milliseconds * 1e-3
+    angular_freq_radps = freq_hz / (2.0 * np.pi)
+
+    result = MAX_ANGLE_RAD * np.vstack([
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds)
+    ]).transpose()
+
+    not_incls = ALL_INDICES - set(included)
+
+    for not_incl in not_incls:
+        print not_incl
+        result[:, not_incl] == 0.0
+
+    return result
+
+
+def all_hold(target_angle):
+    result = np.ones((LENGTH_TICKS, 4))
+    result *= target_angle
+    return result
+
+
+if __name__ == '__main__':
+    from matplotlib import pyplot as plt
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-p', '--plot', action='store_true')
+    args = parser.parse_args()
+
+    tests = {
+        "set_zero": set_zero,
+        "sine_wave_slow": partial(sine_wave_all, freq_hz=0.1),
+        "sine_wave_medium": partial(sine_wave_all, freq_hz=7.0),
+        "sine_wave_fast": partial(sine_wave_all, freq_hz=25.0),
+
+        "02_symmetric": partial(symmetric, included=(0, 2)),
+        "13_symmetric": partial(symmetric, included=(1, 3)),
+
+        "02_antisymmetric": partial(antisymmetric, included=(0, 2)),
+        "13_antisymmetric": partial(antisymmetric, included=(1, 3)),
+
+        "all_max": partial(all_hold, target_angle=MAX_ANGLE_RAD),
+        "all_min": partial(all_hold, target_angle=-MAX_ANGLE_RAD),
```

Comment:
Using partials here seems more indirect than necessary.  

consider `partial(all_hold, target_angle=-MAX_ANGLE_RAD)` -> `generate_yaml(all_hold(target_angle=-MAX_ANGLE_RAD))` .  This new expression has the same number of symbols but does not allow somewhat unholy partials to exist.  Partials unholy due to being harder to reason about.

Reply:
Could reduce code further by simply mapping `generate_yaml` over values after constructing dict.